scripts/fund-address.ts

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env -S yarn --silent run ts-node
+// Used to fund a given address with native and ERC20 tokens up to a set target limit
+
+import yargs from 'yargs/yargs'
+import BN from 'bn.js'
+import { createChainWrapper } from '@hoprnet/hopr-core-ethereum'
+import { expandVars, moveDecimalPoint, Address, Balance, NativeBalance } from '@hoprnet/hopr-utils'
+import { utils } from 'ethers'
+
+const { PRIVATE_KEY } = process.env
+const PROTOCOL_CONFIG = require('../packages/core/protocol-config.json')
+
+function parseGasPrice(gasPrice: string) {
+  const parsedGasPrice = gasPrice.split(' ')
+  if (parsedGasPrice.length > 1) {
+    return Number(utils.parseUnits(parsedGasPrice[0], parsedGasPrice[1]))
+  }
+  return Number(parsedGasPrice[0])
+}
+
+async function getNativeBalance(chain, address: string) {
+  return await chain.getNativeBalance(Address.fromString(address))
+}
+
+async function getERC20Balance(chain, address: string) {
+  return await chain.getBalance(Address.fromString(address))
+}
+
+async function fundERC20(chain, sender: string, receiver: string, targetBalanceStr: string) {
+  const senderBalance = await getNativeBalance(chain, sender)
+  const balance = await getERC20Balance(chain, receiver)
+  const targetBalanceNr = moveDecimalPoint(targetBalanceStr, Balance.DECIMALS)
+  const targetBalance = new Balance(new BN(targetBalanceNr))
+  const diff = targetBalance.sub(balance)
+
+  // stop if sender doesn't have enough funds
+  if (senderBalance.lte(diff)) {
+    console.log(
+      `sender ${sender} has ${senderBalance.toFormattedString()}, not enough to fund ${diff.toFormattedString()} to receiver ${receiver}`
+    )
+    return
+  }
+
+  // stop if receiver has enough funds already
+  if (diff.lte(new Balance(new BN(0)))) {
```

Comment:
Suggestion: is it possible to call the constructor like `new Balance(0)` to simplify things?

Reply:
No, that would required way more input parsing in the constructor.